src/sonic-yang-models/yang-models/sonic-nvgre-tunnel.yang

```diff
@@ -0,0 +1,106 @@
+module sonic-nvgre-tunnel {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-nvgre-tunnel";
+    prefix nvgre;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+    }
+
+    description "NVGRE Tunnel YANG Module for SONiC OS";
+
+    revision 2021-10-31 {
+        description
+            "First Revision";
+    }
+
+    container sonic-nvgre-tunnel {
+
+        container NVGRE_TUNNEL {
+
+            description "NVGRE_TUNNEL part of config_db.json";
+
+            list NVGRE_TUNNEL_LIST {
+
+                key "tunnel_name";
+
+                leaf tunnel_name {
+                    description "NVGRE Tunnel name";
+
+                    type string {
+                        length 1..255;
```

Comment:
Do we have to follow any specific pattern for tunnel name? like we have Etherentxxx for physical interface..

Reply:
Yes, according to RFC - [3.4.  Reserved VSID](https://datatracker.ietf.org/doc/html/rfc7637#section-3.4)

Reply:
No, there is no restriction, the user can assign any name to NVGRE Tunnel.